classes/services/PKPSubmissionService.inc.php

```diff
@@ -421,14 +421,33 @@ public function getProperties($submission, $props, $args = null) {
 					$values[$prop] = $submission->getAbstract(null);
 					break;
 				case 'discipline':
-					$values[$prop] = $submission->getDiscipline(null);
+					$submissionDisciplineDao = DAORegistry::getDAO('SubmissionDisciplineDAO');
+					$values[$prop] = array_filter($submissionDisciplineDao->getDisciplines($submission->getId(), array_keys(\PKPLocale::getAllLocales())));
 					break;
 				case 'subject':
-					$values[$prop] = $submission->getSubject(null);
+					$submissionSubjectDao = DAORegistry::getDAO('SubmissionSubjectDAO');
+					$values[$prop] = array_filter($submissionSubjectDao->getSubjects($submission->getId(), array_keys(\PKPLocale::getAllLocales())));
+					break;
+				case 'keyword':
+					$submissionKeywordDao = DAORegistry::getDAO('SubmissionKeywordDAO');
+					$values[$prop] = array_filter($submissionKeywordDao->getKeywords($submission->getId(), array_keys(\PKPLocale::getAllLocales())));
+					break;
+				case 'agency':
```

Comment:
Since `keyword` and `agency` are new (and not yet [documented for 3.1.x](https://docs.pkp.sfu.ca/dev/api/ojs/3.1)), lets use the plural, `keywords` and `supportingAgencies`. This matches how the properties will be identified in 3.2: https://github.com/NateWr/pkp-lib/blob/i2072_versioning/schemas/publication.json#L136

Reply:
Done!